driver.go

```diff
@@ -755,6 +759,15 @@ func (c *conn) InDMLBatch() bool {
 	return (c.batch != nil && c.batch.tp == dml) || (c.inReadWriteTransaction() && c.tx.(*readWriteTransaction).batch != nil)
 }
 
+func (c *conn) GetBatchedStatements() []spanner.Statement {
+	if c.batch == nil {
+		return []spanner.Statement{}
+	}
+	res := make([]spanner.Statement, len(c.batch.statements))
+	copy(res, c.batch.statements)
+	return res
```

Comment:
Tip: you can use `return slices.Clone(c.batch.statements)` for slightly shorter code.